pkg/agent/containerd/containerd.go

```diff
@@ -110,15 +112,160 @@ func Run(ctx context.Context, cfg *config.Node) error {
 	return PreloadImages(ctx, cfg)
 }
 
+// Watcher is a controller that watch the agent/images folder
+// to ensure that every new file is added to the watcher state
+func Watcher(ctx context.Context, cfg *config.Node) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		logrus.Errorf("Error to create a watcher: %s", err.Error())
+		return
+	}
+
+	// Add agent/images path to the watcher.
+	err = watcher.Add(cfg.Images)
+	if err != nil {
+		logrus.Errorf("Error when creating the watcher controller: %s", err.Error())
+		return
+	}
+
+	client, err := Client(cfg.Containerd.Address)
+	if err != nil {
+		logrus.Errorf("Error to create containerd client: %s", err.Error())
+		return
+	}
+
+	criConn, err := cri.Connection(ctx, cfg.Containerd.Address)
+	if err != nil {
+		logrus.Errorf("Error to create CRI connection: %s", err.Error())
```

Comment:
Error handling here is to just print a message and return, leaving K3s with no watch on the images. Should we have this instead return an error, and wrap it in a retry loop so that it is restarted if it fails?

Also note that the idiomatic error message should be in the rough form of:
```golang
logrus.Errorf("Failed to SOMETHING with %s: %v", thing.Name, err)
```

"Error to" is not proper grammar, and we already know it's an error because we are using `Errorf` which prints `level=error` as part of the log entry.